src/test/javascript/spec/component/programming-exercise/programming-exercise-plans-and-repositories-preview.component.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { AuxiliaryRepository } from 'app/entities/programming-exercise-auxiliary-repository-model';
+import { SimpleChanges } from '@angular/core';
+import { ProgrammingExerciseBuildPlanCheckoutDirectoriesComponent } from 'app/exercises/programming/shared/build-details/programming-exercise-build-plan-checkout-directories.component';
+import { ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent } from 'app/exercises/programming/shared/build-details/programming-exercise-repository-and-build-plan-details.component';
+
+describe('ProgrammingExerciseBuildDetailsComponent', () => {
+    let component: ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent;
+    let fixture: ComponentFixture<ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent>;
+    let programmingExerciseService: ProgrammingExerciseService;
+
+    const CHECKOUT_DIRECTORY_PREVIEW_SUBMISSION_BUILD_PLAN = '#checkout-directory-preview-submission-build-plan';
+    const CHECKOUT_DIRECTORY_PREVIEW_SOLUTION_BUILD_PLAN = '#checkout-directory-preview-solution-build-plan';
+
+    const JAVA_CHECKOUT_DIRECTORIES: CheckoutDirectoriesDto = {
+        submissionBuildPlanCheckoutDirectories: {
+            exerciseCheckoutDirectory: '/assignment',
+            testCheckoutDirectory: '',
+        },
+        solutionBuildPlanCheckoutDirectories: {
+            solutionCheckoutDirectory: '/assignment',
+            testCheckoutDirectory: '',
+        },
+    };
+
+    const OCAML_CHECKOUT_DIRECTORIES: CheckoutDirectoriesDto = {
+        submissionBuildPlanCheckoutDirectories: {
+            exerciseCheckoutDirectory: '/assignment',
+            solutionCheckoutDirectory: '/solution',
+            testCheckoutDirectory: 'tests',
+        },
+        solutionBuildPlanCheckoutDirectories: {
+            solutionCheckoutDirectory: '/assignment',
+            testCheckoutDirectory: 'tests',
+        },
+    };
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            declarations: [ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent, MockComponent(HelpIconComponent), ProgrammingExerciseBuildPlanCheckoutDirectoriesComponent],
+            providers: [{ provide: ProgrammingExerciseService, useClass: MockProgrammingExerciseService }],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(ProgrammingExerciseRepositoryAndBuildPlanDetailsComponent);
+                component = fixture.componentInstance;
+                programmingExerciseService = TestBed.inject(ProgrammingExerciseService);
+
+                component.programmingLanguage = ProgrammingLanguage.C;
+                component.programmingExercise = { id: 1, shortName: 'shortName' } as ProgrammingExercise;
+                component.isLocal = true;
+
+                jest.spyOn(programmingExerciseService, 'getCheckoutDirectoriesForProgrammingLanguage').mockImplementation((programmingLanguage: ProgrammingLanguage) => {
```

Comment:
Do you think it could make sense to include `checkoutSolution: boolean` here too since the component has changed to include this new detail? I suppose you'd have to add another test in that case